src/input-handler.cpp

```diff
@@ -9,8 +9,23 @@ input_handler_t::input_handler_t(osmium::Box const &bbox, bool append,
 : m_data(osmdata), m_bbox(bbox), m_append(append)
 {}
 
+void input_handler_t::negative_id_warning()
+{
+    fmt::print(stderr,
+               "\nWARNING: The input file contains at least one object with "
+               "a negative id.\n         Negative ids are not properly "
+               "supported in osm2pgsql (and never\n         were). They will "
+               "not work in future versions at all. You can use\n"
+               "         the osmium tool to 'renumber' your file.\n\n");
```

Comment:
Can we reformat this text? If necessary, the max line length can be overridden for this part of the file.

Reply:
I have forced pushed a new version with better formatting.